### PR TITLE
PanelEdit: Allow standard field config properties extensions without altering order of standard options 

### PR DESCRIPTION
**Pull request by dprokop**
With "Show thresholds as..." option introduced in the Time series Panel, the Thresholds and Standard options categories got switched (thanks @oddlittlebird for catching this):

image

This PR mitigates this unwanted behaviour making sure that the custom options added to standard categories do no change the order of common options.

Also - this PR fixes Tooltip config being rendered under Legend category.

### CODE REVIEW

File: packages/grafana-data/src/panel/registryFactories.ts [hunk 1]
```diff
@@ -58,7 +67,19 @@ export function createFieldConfigRegistry<TFieldConfigOptions>(
...
+      }
+    }
   }
 
   return registry;
 }
+
+function isStandardConfigExtension(property: FieldConfigPropertyItem, standardProperties: FieldConfigPropertyItem[]) {
+  return Boolean(
+    standardProperties.find((p) => property.category && p.category && property.category[0] === p.category[0])
```
Comment (dprokop) [inline packages/grafana-data/src/panel/registryFactories.ts:83]: Thinking about deprecating category as an array - seems like we are not using it anywhere this way, always a single string. @torkelo do you recall any nested categories somewhere?

**Review by torkelo — COMMENTED** [review]
I think we need a different solution so plugins can control the order of categories. For example want Legend to be after Graph Styles category in Time series. But not possible as Legend category only contains panel level settings, and currently panel level settings are processed first when building display categories. 

If we can somehow specify order that could potentially solve this as well?

But how to specify order....

File: packages/grafana-data/src/panel/registryFactories.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-data/src/panel/registryFactories.ts:83]: don't think so, also been thinking of making this string, and an id. And elsewhere be able to add category info objects with id, name, description & sort order . With a category info object we could more easily control sorting, add description, and maybe help links & tips

**Review by torkelo — APPROVED** [review]

**Comment by dprokop** [comment]
@torkelo and me decided to merge this fix and do a followup improvements to categories definition in https://github.com/grafana/grafana/issues/34789

**Pull request merged**